src/Esprima/ArrayList.cs

```diff
@@ -348,16 +327,6 @@ public void Yield(out T[]? items, out int count)
         this = default;
     }
 
-    public void TrimExcess(int threshold = MinAllocatedCount)
```

Comment:
The poor thing didn't live long... :(

I added it not to improve performance but to reduce memory footprint in cases where we build a list with `ArrayList<T>`, then hand it over to the consumer via a public API as a `ReadOnlySpan<T>` (to avoid copying the list in lucky cases where the underlying array is almost full, that is, doesn't have many unused items).

For example, in the case of collecting comments in the scanner, it could happen that there are 1025 comments in the scanned script, so the underlying array's capacity will increase to 2048. Which means that there will be 1023 unused items, which will be kept around in the memory since we return the underlying array via a span.

`TrimExcess` was introduced to improve this situation, so I'd still keep it.

Reply:
Reverted the removal.